compiler/x/codegen/X86SystemLinkage.cpp

```diff
@@ -429,7 +429,7 @@ TR::X86SystemLinkage::createPrologue(TR::Instruction *cursor)
 
    const TR::X86LinkageProperties &properties = getProperties();
 
-   const uint32_t outgoingArgSize = cg()->getLargestOutgoingArgSize();
+   const uint32_t outgoingArgSize = (cg()->isLeafMethod()) ? 0 : cg()->getLargestOutgoingArgSize();
```

Comment:
My reading of the code wherever `_largestOutgoingArgSize` is set leads me to believe it should only be non-zero when a call with arguments is present (and hence this shouldn't be a leaf method).  I would prefer it being removed if not necessary as it might add confusion in the future.